### PR TITLE
Extract Full Table Names for both the `from_tables` and `target_tables` column.

### PR DESCRIPTION
**Pull request by ericxiao251**
Any function that added tables to the `fromTables` and `targetTable` list, we want to get the full table names, not just the `suffix`. We can do this by calling `toString()` on the `node` variable.

I updated the tests according and intentionally left a test without any `catalog` / `schema` values.

Question: Should we also look to extract full table names for both the `CTE` table names (`cteTables`) and aliased table names (`aliasTables`) in queries? I wasn't aware that you could add a `catalog` or `schema` to those values in a query (if it is indeed possible).

### CODE REVIEW

File: src/test/java/com/shopify/presto/eventlistener/TestQueryDetails.java [hunk 1]
```diff
@@ -12,8 +12,18 @@
...
+        assertEquals(newHashSet("hive.scratch.t1"), d.getFromTables());
+        assertEquals(Collections.emptySet(), d.getAliasTables());
+        assertEquals("", d.getTargetTable());
+        assertEquals(Collections.emptySet(), d.getCTETables());
+        assertEquals("Query", d.getOperation());
+    }
+
+    @Test
+    public void testParseTablesWithQuotes() {
+        QueryDetails d = QueryDetails.parseQueryDetails("SELECT * FROM hive.\"scratch\".\"t1\"");
```
Comment (julienatshopify) [inline src/test/java/com/shopify/presto/eventlistener/TestQueryDetails.java]: What happens when the quoted schema or table name is a SQL keyword, for example "select"?

**Comment by batareyka** [comment]
I also don't think you can add catalogue or schema to alias or cte tables 🤷‍♀

**Comment by batareyka** [comment]
I was also hoping that if it is coming directly from Presto, we would be able to know schemas and catalogues of tables when the query text doesn't specify them 🤔 when people set their catalogue.schema for the session, they can omit them in query text, but Presto would still know what catalogue/schema to use. Any way we can get that info?

**Comment by ericxiao251** [comment]
> Something that would help a lot is a field indicating the event is for a query or something else (ex: CREATE VIEW). It would make filtering very efficient downstream, and not make us guess what happens. Do you think a follow up PR could take care of that?

I believe the `query_operation` column and `query_target_table` would help with this?

So we group together the `views` and `tables` created in a presto query into the `query_target_table` set. Maybe we could break this out into two separate sets?

**Comment by ericxiao251** [comment]
> I was also hoping that if it is coming directly from Presto, we would be able to know schemas and catalogues of tables when the query text doesn't specify them 🤔 when people set their catalogue.schema for the session, they can omit them in query text, but Presto would still know what catalogue/schema to use. Any way we can get that info?

Without looking at the code too in-depth, I'm not 100% confident that this might be picked up. It looks like we only pass in the sql string and parse the sql string. The session that sets the catalog and schema happens before this, so we would need to look into more details on how to go about that.

Let's just ship this change and document all the things we might want to add to this dataset? ie. catalog, schema values from session level. views, tables created, etc.

**Comment by ericxiao251** [comment]
Thinking out loud: perhaps we could add a `SessionDetails` class that would capture all the session parameters people set, then add those to the event. Some interesting findings could come out of this, ie. what are people setting in terms of optimization parameters, etc.

**Comment by ericxiao251** [comment]
So it's definitely possible to get (all?) the session level values (look below at the snippet), but maybe we should regroup to see how we want to implement this back into the `from_tables` `target_tables` column and what else we want to grab from the context.

Screen Shot 2019-10-28 at 4 23 58 PM

**Comment by ericxiao251** [comment]
@MatthewRBruce,  To deploy, we just need to follow the steps in the readme of this repo?

**Comment by MatthewRBruce** [comment]
That will push it to PackageCloud, but we will have to rebuild the presto containers (https://github.com/Shopify/data-docker-images/blob/master/presto-container/Dockerfile#L36)  and re-deploy the cluster to actually get it live.

**Comment by ericxiao251** [comment]
Ahhh yes, right I remember now. When do you think we can do that, I know we recently just did one.

**Comment by MatthewRBruce** [comment]
Depends on the urgency.  If it's just a 'nice to have' then probably a couple of weeks.  If there's a good reason, we could probably :tophat: it this week and deploy it either late this week or early next week

**Comment by ericxiao251** [comment]
It's pretty urgent, most of the downstream models we are trying to reverse engineer the table names from the `query_text` column, ie. the data experience presto/mode tables. As well as the ex-refinery presto queries model, we can't have the `query_text` column as it could contain PII...

**Comment by julienatshopify** [comment]
Re: about urgency, the solution we have now is good enough for us. We can wait a couple of weeks on our end.

**Pull request merged**